consensus/ibft/src/test/java/tech/pegasys/pantheon/consensus/ibft/SynchronizerUpdaterTest.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.consensus.ibft;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import tech.pegasys.pantheon.ethereum.eth.manager.ChainState;
+import tech.pegasys.pantheon.ethereum.eth.manager.EthPeer;
+import tech.pegasys.pantheon.ethereum.eth.manager.EthPeers;
+import tech.pegasys.pantheon.ethereum.p2p.api.MessageData;
+import tech.pegasys.pantheon.ethereum.p2p.api.PeerConnection;
+import tech.pegasys.pantheon.ethereum.p2p.wire.Capability;
+import tech.pegasys.pantheon.ethereum.p2p.wire.PeerInfo;
+import tech.pegasys.pantheon.ethereum.p2p.wire.messages.DisconnectMessage.DisconnectReason;
+
+import java.net.SocketAddress;
+import java.util.Set;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SynchronizerUpdaterTest {
+
+  @Mock private EthPeers ethPeers;
+  @Mock private EthPeer ethPeer;
+
+  @Mock private ChainState chainState;
+
+  @Test
+  public void ethPeerIsMissingResultInNoUpdate() {
+    when(ethPeers.peer(any())).thenReturn(null);
+
+    final SynchronizerUpdater updater = new SynchronizerUpdater(ethPeers);
+
+    updater.updatePeerChainState(1, createAnonymousPeerConnection());
+
+    verifyZeroInteractions(ethPeer);
+  }
+
+  @Test
+  public void chainStateIsNotUpdatedIfItIsAheadOfRequestedHeight() {
+    final long knownChainHeight = 5L;
+    final long suppliedChainHeight = 4L;
+    when(ethPeers.peer(any())).thenReturn(ethPeer);
+    when(ethPeer.chainState()).thenReturn(chainState);
+    when(chainState.getEstimatedHeight()).thenReturn(knownChainHeight);
+
+    final SynchronizerUpdater updater = new SynchronizerUpdater(ethPeers);
+
+    updater.updatePeerChainState(suppliedChainHeight, createAnonymousPeerConnection());
+
+    verify(chainState, times(0)).updateHeightEstimate(anyLong());
```

Comment:
Should the `anyLong()` be `knownChainHeight`? ...unless any value is really acceptable?

Reply:
given this is a verify (rather than a "thenReturns") - I'd like to make sure that this function wasn't invoked at all - regardless of the parameter.